### ISSUE TITLE
files and folders cannot be renamed after Acronis True Image is installed

### ISSUE THREAD

**Issue by heldchen** [microsoft/WSL#2451]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
16273

* What you're doing and what's happening:  (Copy&paste specific commands and their output, or include screen shots)

after updating Acronis True Image 2017 to Acronis True Image 2018 (Build 9207), files within WSL cannot be renamed anymore:
```
cd /tmp
touch test
mv test test2
```
killing the `Acronis Drive Service` process in task manager makes `mv` work again.

* What's wrong / what should be happening instead:
`mv: cannot move 'test' to a subdirectory of itself, 'test2'`

* Strace of the failing command, if applicable:  (If `<cmd>` is failing, then run `strace -o strace.txt -ff <cmd>`, and post the strace.txt output here)
 
`rename("test", "test2")                 = -1 EINVAL (Invalid argument)`

```
execve("/bin/mv", ["mv", "test", "test2"], [/* 29 vars */]) = 0
brk(NULL)                               = 0x7ffff6020000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap(NULL, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fd4e01b0000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=57226, ...}) = 0
mmap(NULL, 57226, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fd4e01a2000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libselinux.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0pa\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=154904, ...}) = 0
mmap(NULL, 2259152, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4dfbd0000
mprotect(0x7fd4dfbf5000, 2093056, PROT_NONE) = 0
mmap(0x7fd4dfdf4000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x24000) = 0x7fd4dfdf4000
mmap(0x7fd4dfdf6000, 6352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fd4dfdf6000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libacl.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20\34\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=31232, ...}) = 0
mmap(NULL, 2126336, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4df9c0000
mprotect(0x7fd4df9c7000, 2093056, PROT_NONE) = 0
mmap(0x7fd4dfbc6000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6000) = 0x7fd4dfbc6000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libattr.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\300\20\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=18624, ...}) = 0
mmap(NULL, 2113760, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4df7b0000
mprotect(0x7fd4df7b4000, 2093056, PROT_NONE) = 0
mmap(0x7fd4df9b3000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3000) = 0x7fd4df9b3000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20\5\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1856752, ...}) = 0
mmap(NULL, 3959200, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4df3e0000
mprotect(0x7fd4df59e000, 2093056, PROT_NONE) = 0
mmap(0x7fd4df79d000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1bd000) = 0x7fd4df79d000
mmap(0x7fd4df7a3000, 14752, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fd4df7a3000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libpcre.so.3", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340\24\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=464824, ...}) = 0
mmap(NULL, 2560264, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4df160000
mprotect(0x7fd4df1d1000, 2093056, PROT_NONE) = 0
mmap(0x7fd4df3d0000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x70000) = 0x7fd4df3d0000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\200\r\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=14608, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fd4e01a0000
mmap(NULL, 2109680, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4def50000
mprotect(0x7fd4def53000, 2093056, PROT_NONE) = 0
mmap(0x7fd4df152000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7fd4df152000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\240`\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=142400, ...}) = 0
mmap(NULL, 2217000, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fd4ded30000
mprotect(0x7fd4ded48000, 2097152, PROT_NONE) = 0
mmap(0x7fd4def48000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x18000) = 0x7fd4def48000
mmap(0x7fd4def4a000, 13352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fd4def4a000
close(3)                                = 0
mmap(NULL, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fd4e0190000
arch_prctl(ARCH_SET_FS, 0x7fd4e0190800) = 0
mprotect(0x7fd4df79d000, 16384, PROT_READ) = 0
mprotect(0x7fd4def48000, 4096, PROT_READ) = 0
mprotect(0x7fd4df152000, 4096, PROT_READ) = 0
mprotect(0x7fd4df3d0000, 4096, PROT_READ) = 0
mprotect(0x7fd4df9b3000, 4096, PROT_READ) = 0
mprotect(0x7fd4dfbc6000, 4096, PROT_READ) = 0
mprotect(0x7fd4dfdf4000, 4096, PROT_READ) = 0
mprotect(0x7fd4e041d000, 4096, PROT_READ) = 0
mprotect(0x7fd4e0025000, 4096, PROT_READ) = 0
munmap(0x7fd4e01a2000, 57226)           = 0
set_tid_address(0x7fd4e0190ad0)         = 14055
set_robust_list(0x7fd4e0190ae0, 24)     = 0
rt_sigaction(SIGRTMIN, {0x7fd4ded35b40, [], SA_RESTORER|SA_SIGINFO, 0x7fd4ded41670}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {0x7fd4ded35bd0, [], SA_RESTORER|SA_RESTART|SA_SIGINFO, 0x7fd4ded41670}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
statfs("/sys/fs/selinux", 0x7ffffe8bada0) = -1 ENOENT (No such file or directory)
statfs("/selinux", 0x7ffffe8bada0)      = -1 ENOENT (No such file or directory)
brk(NULL)                               = 0x7ffff6020000
brk(0x7ffff6041000)                     = 0x7ffff6041000
open("/proc/filesystems", O_RDONLY)     = 3
fstat(3, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
read(3, "nodev\tsysfs\nnodev\trootfs\nnodev\tb"..., 4096) = 312
read(3, "", 4096)                       = 0
close(3)                                = 0
access("/etc/selinux/config", F_OK)     = -1 ENOENT (No such file or directory)
open("/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=2011424, ...}) = 0
mmap(NULL, 2011424, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fd4dfe39000
close(3)                                = 0
geteuid()                               = 0
ioctl(0, TCGETS, {B38400 opost isig icanon echo ...}) = 0
stat("test2", 0x7ffffe8bad70)           = -1 ENOENT (No such file or directory)
lstat("test", {st_mode=S_IFREG|0644, st_size=0, ...}) = 0
lstat("test2", 0x7ffffe8baa50)          = -1 ENOENT (No such file or directory)
rename("test", "test2")                 = -1 EINVAL (Invalid argument)
open("/usr/share/locale/locale.alias", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=2995, ...}) = 0
read(3, "# Locale name alias data base.\n#"..., 512) = 512
read(3, "See the\n# GNU General Public Lic"..., 512) = 512
read(3, "nd for the time being for\n# back"..., 512) = 512
read(3, "hese lines were removed\n# becaus"..., 512) = 512
read(3, "8859-1\ngalego\t\tgl_ES.ISO-8859-1\n"..., 512) = 512
read(3, "O.ISO-8859-1 nb_NO.ISO-8859-1\nno"..., 512) = 435
read(3, "", 512)                        = 0
close(3)                                = 0
open("/usr/share/locale/en_US/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en_US/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
write(2, "mv: ", 4)                     = 4
write(2, "cannot move 'test' to a subdirec"..., 55) = 55
write(2, "\n", 1)                       = 1
lseek(0, 0, SEEK_CUR)                   = -1 ESPIPE (Illegal seek)
close(0)                                = 0
close(1)                                = 0
close(2)                                = 0
exit_group(1)                           = ?
+++ exited with 1 +++

```

**Comment by heldchen** [comment]
rename within mounted drives works, but every rename action within rootfs fails...

**Comment by heldchen** [comment]
Acronis build 9207 has some kind of protection service included, which previous versions do not seem to have: https://www.acronis.com/en-us/support/updates/changes.html?p=40697  - turning off this feature does not help with the rename problem within WSL tho.

**Comment by sunilmut** [comment]
Bringing this to @SvenGroot attention to see how the "True Image" might be interfering with the `rename` operation. I wonder if they use some kind of filesystem filter driver. There are differences in the way `rename` happen within DrvFs (i.e. /mnt/c) and LxFS (/ or ~), which could explain the difference you are observing above.

**Comment by heldchen** [comment]
after further testing the problem is caused by the `Acronis Drive Service` process, which seems to be used for file mounting. killing the process fixes the WSL problems.

**Comment by shevegen** [comment]
Very interesting - seems to be a way to have immortal directories ... :D

PS: I dig your nick "heldchen".

> killing the process fixes the WSL problems.

I guess it depends which processes should have priority. The assumption may
actually be correct that Acronis or xyz would be able to prevent file operations,
but I can totally understand that this can be frustrating, in particular when you
may not know what is preventing this from happening (in this case third party
applications).

**Comment by heldchen** [comment]
in the case of the "Acronis Drive Service", this helper tool only provides some sort of virtual drive that contains files one might have put into the "Acronis Cloud". when the process is run, it seems to synchronize a folder within the user's profile (`C:\Users\MyName\AppData\Local\Acronis\Acronis Drive`) with their cloud, set a `subst` to this folder for drive `M:`, and adds a network location shortcut in Windows File Explorer. 

![image](https://user-images.githubusercontent.com/299636/29748496-467d4a42-8b18-11e7-887d-25e155a3f2a9.png)

to me, this service looks completely unrelated to the actual acronis file backup services monitoring file changes (i.e. for sector based backups) - in fact all backup & restore operations seem to continue to run just fine after I prevented the execution of said helper exe (see ps1 script here: https://github.com/Microsoft/BashOnWindows/issues/2437#issuecomment-325058101). but then again, I do not use the Archive / Sync features using their cloud services...

to me this looks like an unwanted side effect or even bug within the Acronis Drive Service, maybe it hasn't setup its file monitors to limit them to its AppData folder. I've reached out to the acronis support but based on past support experiences with them I don't have high hopes.

> I dig your nick "heldchen".

thanks. it's the handle from the  MSN dialup beta testing days in 1995... :-)
